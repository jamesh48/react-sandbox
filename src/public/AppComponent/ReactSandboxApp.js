import React from 'react';
import './app.scss';

export default ({ children }) => {

  const [selectedComponentId, setSelectedComponentId] = React.useState('');

  const handleSelectChange = () => {
    setSelectedComponentId(event.target.value);
  };

  return (
    <div id='app-root'>

      <select id='app-select' onChange={handleSelectChange}>
        <option value='default'>Select a component...</option>
        {children?.map(({ key, props: { label } }, index) => {
          return <option key={key} value={index}>{label}</option>
        })}
      </select>

      <main>
        {selectedComponentId ? children[selectedComponentId] : null}
      </main>
    </div>
  )
}