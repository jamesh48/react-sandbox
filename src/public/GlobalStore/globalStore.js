import React from 'react';
import CombinedReducers from './globalReducers.js';
const GlobalStoreContext = React.createContext();

const initialState = {};

const GlobalStoreProvider = ({ children }) => {
  const [state, dispatch] = React.useReducer(CombinedReducers,
    initialState);
  const store = React.useMemo(() => [state, dispatch], [state]);
  return (
    <GlobalStoreContext.Provider value={store}>{children}</GlobalStoreContext.Provider>
  );
};

const useGlobalContext = () => {
  const context = React.useContext(GlobalStoreContext);
  if (context === undefined) {
    throw new Error('useGlobalContext must be used within a GlobalProvider')
  }
  return context;
};

export { useGlobalContext, GlobalStoreProvider };