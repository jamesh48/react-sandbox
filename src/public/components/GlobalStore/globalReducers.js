// import {  } from './globalActions.js';

const combineReducers = (slices) => (state, action) => Object.keys(slices).reduce((acc, prop) => ({
  ...acc,
  [prop]: slices[prop](acc[prop], action),
}),
  state
);

const reduceReducers = (...reducers) => {
  return (state, action) => reducers.reduce((acc, nextReducer) => nextReducer(acc, action), state)
};

export default combineReducers({});