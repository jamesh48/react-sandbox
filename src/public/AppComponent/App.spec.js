import React from 'react';
import renderer from 'react-test-renderer';
import ReactSandboxApp from './ReactSandboxApp.js';
import { GlobalStoreProvider } from 'GlobalStore';

import { mount } from 'enzyme';

describe('ReactSandboxApp Component', () => {

  it('Renders the ReactSanboxApp Component', () => {
    const wrapper = mount(
      <GlobalStoreProvider>
        <ReactSandboxApp />
      </GlobalStoreProvider>
    );
    expect(wrapper.find('#app-root').length).toEqual(1);
  })

  test('snapshot renders', () => {
    const component = renderer.create();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});