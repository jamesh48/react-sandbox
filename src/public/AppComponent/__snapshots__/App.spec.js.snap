// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactSandboxApp Component snapshot renders 1`] = `null`;
